backend/models/events/events-model.test.js

```diff
@@ -11,7 +22,95 @@ afterAll(async () => await db.closeDatabase())
 
 // Add your test here
 describe("Events Model Tests", () => {
-    test("Dummy test", () => {
-        expect(true).toBe(true)
+    test("Database Has Events Model", () => {
+        expect(EventsModel).toBeDefined()
+    })
+
+    describe("Event Test Series", () => {
+        test("Get Event Instance from Database", async () => {
+            const event = new EventsModel(testEvent1)
+            await event.save()
+            const retrievedEvent = await EventsModel.findOne({
+                title: "Hackathon",
+            })
+
+            // Expected value set
+            const expectedTitle = testEvent1.title
+            const expectedDetail = testEvent1.detail
+            const expectedLocation = testEvent1.location
+            const expectedDate = testEvent1.date
+            const expectedCreatedDate = testEvent1.createdDate
+            const expectedUpdatedDate = testEvent1.updatedDate
+
+            // Tests
+            expect(expectedTitle).toEqual(retrievedEvent.title)
+            expect(expectedDetail).toEqual(retrievedEvent.detail)
+            expect(expectedLocation).toEqual(retrievedEvent.location)
+            expect(expectedDate).toEqual(retrievedEvent.date)
+            expect(expectedCreatedDate).toEqual(retrievedEvent.createdDate)
+            expect(expectedUpdatedDate).toEqual(retrievedEvent.updatedDate)
+        })
+
+        test("Add an Event to Database", async () => {
+            const event = new EventsModel(testEvent1)
+            const retrievedEvent = await event.save()
```

Comment:
Why is retrievedEvent different here vs line 33?

Reply:
Line 33 uses findOne() to get the instance successfully from the database. 
Line 56 assigns the copy of object if it was created successfully in the database, which is a promise, not retrieval.